mage_ai/io/google_sheets.py

```diff
@@ -0,0 +1,289 @@
+        the same parameters as a service key) with the keyword argument `credentials_mapping`
+
+        All keyword arguments except for `path_to_credentials` and `credentials_mapping` will be
+        passed to the Google Sheet client, accepting all other configuration settings there.
+        """
+
+        super().__init__(verbose=kwargs.get('verbose', True))
+
+        if kwargs.get('verbose') is not None:
+            kwargs.pop('verbose')
+
+        scopes = [
+            'https://www.googleapis.com/auth/spreadsheets',
+            'https://www.googleapis.com/auth/drive',
+        ]
+
+        credentials = kwargs.get('credentials')
+        if credentials is None:
+            if 'credentials_mapping' in kwargs:
+                mapping_obj = kwargs.pop('credentials_mapping')
+                if mapping_obj is not None:
+                    credentials = service_account.Credentials.from_service_account_info(
+                        mapping_obj, scopes=scopes
+                    )
+            if 'path_to_credentials' in kwargs:
+                path = kwargs.pop('path_to_credentials')
+                if path is not None:
+                    credentials = service_account.Credentials.from_service_account_file(
+                        path, scopes=scopes
+                    )
+            if 'credentials' in kwargs:
+                kwargs.pop('credentials')
+        with self.printer.print_msg('Opening connection to Google Sheets'):
+            self.client = gspread.authorize(credentials, **kwargs)
+
+    def fetch_sheet(
+        self,
+        sheet_url: str = None,
+        sheet_id: str = None,
+        sheet_name: str = None,
+    ) -> gspread.spreadsheet.Spreadsheet:
+        """Fetches a Google Sheet, given a sheet_url, sheet_id, or sheet_name.
+
+        Args:
+            sheet_url (str, optional): A sheet URL to export the dataframe to. Defaults to None
+            sheet_id (str, optional): A sheet id to export the dataframe to. Defaults to None
+            sheet_name (str, optional): A sheet name (title) to export the df. Defaults to None
+
+        Raises:
+            KeyError: _description_
+
+        Returns:
+            gspread.spreadsheet.Spreadsheet: A gspread Spreadsheet object.
+        """
+        sheet = None
+
+        if sheet_url:
+            sheet = self.client.open_by_url(sheet_url)
+        elif sheet_id:
+            sheet = self.client.open_by_key(sheet_id)
+        elif sheet_name:
+            sheet = self.client.open(sheet_name)
+        else:
+            raise KeyError(
+                'Please set one of sheet_url, sheet_id, or sheet_name to a valid Google Sheet.'
+            )
+        if sheet is not None:
+            with self.printer.print_msg(
+                f'Loaded Google Sheet \'{sheet.title}\' with ID \'{sheet.id}\''
+            ):
+                return sheet
+        else:
+            raise ValueError('There was an error loading the Google Sheet.')
```

Comment:
same here, you can just return None, and check for None when calling this method. As a developer, I think I'd rather know that it returned None than just a vague error message.